openpype/hosts/houdini/plugins/publish/extract_bgeo.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+import pyblish.api
+
+from openpype.pipeline import publish
+from openpype.hosts.houdini.api.lib import render_rop
+
+import hou
+
+
+class ExtractBGEO(publish.Extractor):
+
+    order = pyblish.api.ExtractorOrder
+    label = "Extract BGEO"
+    hosts = ["houdini"]
+    families = ["bgeo"]
+
+    def process(self, instance):
+
+        ropnode = hou.node(instance.data["instance_node"])
+
+        # Get the filename from the filename parameter
+        output = ropnode.evalParm("sopoutput")
+        staging_dir = os.path.dirname(output)
+        instance.data["stagingDir"] = staging_dir
+
+        file_name = os.path.basename(output)
+
+        # We run the render
+        self.log.info("Writing bgeo files '%s' to '%s'" % (file_name,
+                                                        staging_dir))
+
+        # write files
+        ropnode.parm("execute").pressButton()
```

Comment:
Actually there was a reason as I've initially used `render_rop()`. And it didn't work for me, complaining that there is no `render()` method on the node. But since I've restarted everything it works again. Some dark Houdini magic there.